### ISSUE TITLE
error message: couldn't make stderr distinct from stdout

### ISSUE THREAD

**Issue by Amleto** [Civitasv/cmake-tools.nvim#105]
**Bug description**
when cmake-tools starts, the message `dtable::stdio_init couldn't make stderr distinct from stdout` flashes up.

**Steps to reproduce**
Config:
```
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    "--branch=stable", -- latest stable release
    lazypath,
  })
end

vim.opt.rtp:prepend(lazypath)


local plugins = {

  -- cmake
  {
    'Civitasv/cmake-tools.nvim',
    lazy = false,
    config = function(plugin) require('cmake-tools').setup({}) end,
    cmd = {'CMakeBuild', 'CMakeGenerate'},
    -- keys = {
    --   {"<leader>cc", ":CMakeBuild<CR>", desc = "CMake Configure"},
    -- },
    dependencies = {
      'nvim-lua/plenary.nvim',
    },
  },
}

local opts = {}
require("lazy").setup(plugins, opts)
```

Start nvim (that's it)

**Expected behavior**
At best, no error message. Improvement would be the error being reported to a log with a call stack.

**Screenshots**
![signal-2023-07-29-114428_002](https://github.com/Civitasv/cmake-tools.nvim/assets/2996796/ce539395-689a-4ccd-8d5f-228cbb6364fe)

**Environment**

- OS: Win10
- Terminal: win terminal or wezterm
- shell: git-bash

**Comment by Civitasv** [comment]
Can this solve your problem: https://stackoverflow.com/questions/11868017/couldnt-make-stderr-distinct-from-stdout-when-running-cygwin-commands

**Comment by Amleto** [comment]
What, exactly?

**Comment by Civitasv** [comment]
Can’t you see the stackoverflow link?

**Comment by Amleto** [comment]
Yes, what part of it? I don't have cygwin and there are multiple aspects in that link so I don't know what you're referring to.

**Comment by Civitasv** [comment]
Okay, I understand. But I don’t know if this problem is related to cmake-tools.nvim. The error message is not very specific.

**Comment by Amleto** [comment]
Ok, I understand it might not be from cmake-tools, but something is still going wrong and not logged.

**Comment by Civitasv** [comment]
Yes. Maybe a window’s issue.

**Comment by Amleto** [comment]
On windows should cmake-tools even be trying to use 'touch'?

**Comment by Civitasv** [comment]
> On windows should cmake-tools even be trying to use 'touch'?

No, it doesn’t. I mean this maybe a issue of neovim on Windows.

**Comment by Amleto** [comment]
whats [this](https://github.com/Civitasv/cmake-tools.nvim/blob/ffbc67d283b3277c07bfa6029223e1b0fcec6f7c/lua/cmake-tools/session.lua#L42C3-L42C3)?

**Comment by Civitasv** [comment]
Oh, sorry for forgetting this. It uses this to generate session files.

**Comment by Civitasv** [comment]
Then maybe that’s the reason. Can you test it by removing this line?

**Comment by Amleto** [comment]
I can verify that _is_ the cause of the error message.

**Comment by Civitasv** [comment]
Interesting, i wonder if it will popup error if you directly execute the touch command. Maybe the reason is that users don’t have write permission to that directory?

**Comment by Amleto** [comment]
I made pull request:

#107

**Comment by Amleto** [comment]
I closed the PR due to the exception seen (attached in the PR).

There needs to be more changes - I didn't check that the supplied 'path' variable would be correct, which it seems not to be - I added a debug print,

```
local function init_session()
  init_cache()

  local path = get_current_path()
  if not utils.file_exists(path) then
    local file = io.open(path, 'w')
    if file then
      file:close()
    else
      print("file not opened " .. path)
    end
  end
end
```
 and it revealed this:

`file not opened C:\Users\<username>/AppData/Local/cmake_tools_nvim/D:\Coding\C++\Projects\simple-cmake.lua`

**Comment by Amleto** [comment]
So I think this needs adjusting as well

```
local function get_current_path()
  local current_path = vim.loop.cwd()
  local clean_path = current_path:gsub("/", "")
  return get_cache_path() .. clean_path .. ".lua"; -- <<<<
end

``

**Comment by Civitasv** [comment]
Yes, `clean_path` should be properly handled for each OS.

**Comment by Amleto** [comment]
ok, PR updated. Lets try again ^_^

**Comment by Civitasv** [comment]
Fixed by #107.
